### PR TITLE
fix: windows.get_active_window()

### PR DESCRIPTION
**Pull request by jesicasusanto**
**What kind of change does this PR introduce?**

This PR changes the _windows.get_active_window_state() in https://github.com/OpenAdaptAI/OpenAdapt/blob/main/openadapt/window/_windows.py#L31 , to get the top window of the active application. This PR resolves getting the "No windows of that application are active" logger wawrning.

**Summary**

This PR is a bugfix for #331.

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have perfomed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

Create a new recording on windows, switch to a different application. You should not see the "No windows of that application are active" warning again.

### CODE REVIEW

**Comment by jesicasusanto** [comment]
Result of bugfix : 

https://github.com/OpenAdaptAI/OpenAdapt/assets/87709055/34169a10-e21a-4fa6-b3de-0643e46f9a80

File: openadapt/window/_windows.py [hunk 1]
```diff
@@ -25,11 +25,8 @@ def get_active_window_state() -> dict:
                 - "window_id": ID of the active window.
     """
     # catch specific exceptions, when except happens do log.warning
-    try:
-        active_window = get_active_window()
-    except RuntimeError as e:
-        logger.warning(e)
-        return {}
+    active_window = get_active_window()
```
Comment (abrichr) [inline openadapt/window/_windows.py:28]: Please handle exception 🙏

File: openadapt/window/_windows.py [hunk 2]
```diff
@@ -28,7 +28,6 @@ def get_active_window_state() -> dict:
     try:
         active_window = get_active_window()
     except RuntimeError as e:
-        logger.warning(e)
```
Comment (abrichr) [inline openadapt/window/_windows.py:31]: Thanks @jesicasusanto !

Can you please clarify why this was removed?

**Comment by jesicasusanto** [comment]
- [x] address `*** _ctypes.COMError: (-2147467263, 'Not implemented', (None, None, None, 0, None))` error
This error is reproduced by getting the selection_indices() property from a <uia_controls.EditWrapper - 'Address and search bar', Edit, -4439357938799333343> element in https://github.com/OpenAdaptAI/OpenAdapt/pull/333/commits/3dc81e5c9eb058b748e6febb6f60510e03abdd0e
```
ipdb> getattr(element, propname)()
*** _ctypes.COMError: (-2147467263, 'Not implemented', (None, None, None, 0, None))
ipdb> element
<uia_controls.EditWrapper - 'Address and search bar', Edit, -4439357938799333343>
ipdb> propname
'selection_indices'
```

- [x] address ` *** NotImplementedError: This method not work properly for WinForms DataGrid, use cells()`
ipdb> getattr(element, propname)() in https://github.com/OpenAdaptAI/OpenAdapt/pull/333/commits/3dc81e5c9eb058b748e6febb6f60510e03abdd0e
```
*** NotImplementedError: This method not work properly for WinForms DataGrid, use cells()
ipdb> element
<uia_controls.ListViewWrapper - '', Table, -2863142914305943712>
ipdb> propname
'column_count'
```

File: openadapt/window/_windows.py [hunk 3]
```diff
@@ -143,6 +138,37 @@ def dictify_rect(rect):
...
+
+    Returns:
+        A dictionary containing the writable properties of the element, 
+        with property names as keys and their corres
+        ponding values.
+
+    """
+    _element_class = element.__class__
+    class TempElement(element.__class__):
+            writable_props = pywinauto.base_wrapper.BaseWrapper.writable_props
```
Comment (abrichr) [inline openadapt/window/_windows.py:161]: Please fix indent

File: openadapt/window/_windows.py [hunk 4]
```diff
@@ -143,6 +138,37 @@ def dictify_rect(rect):
...
+
+    # Instantiate the subclass
+    element.__class__ = TempElement
+    # Retrieve properties using get_properties()
+    properties = element.get_properties()
+    element.__class__ = _element_class
+    return properties
+
+
+
```
Comment (abrichr) [inline openadapt/window/_windows.py:171]: Please remove extra newline

Comment (0dm) [inline openadapt/window/_windows.py]:
```suggestion
```

Comment (jesicasusanto) [reply openadapt/window/_windows.py]: addressed in https://github.com/OpenAdaptAI/OpenAdapt/pull/333/commits/6685c3695ad5d69a6379f537a55bb7a93b631d39

**Pull request merged**